client/babel.config.js

```diff
@@ -0,0 +1,15 @@
+/**
```

Comment:
Was there a reason to rename this file + change some of the lines?

Reply:
Yes, I mentioned in the PR description as well as in the commit body for that specific change:
> fix babel config as suggested in official docs - the suggested way is to use next/babel preset which has all transforms NextJS needs including the TS transform. Renamed to babel.config.js and added a link to official docs for future reference

But If you need more context:
- The `next/babel` preset is sort of like a [bundle preset](https://github.com/vercel/next.js/blob/canary/packages/next/build/babel/preset.ts) which already includes built-in presets `preset-env` and `preset-react` as well as `preset-typescript` (which this integration needs) and it's the recommended way as suggested in the official [docs](https://nextjs.org/docs/advanced-features/customizing-babel-config). Besides those 3 presets, they include some [custom transforms and plugins](https://github.com/vercel/next.js/blob/426331883638911dec6f93517d6b5406abac1d97/packages/next/build/babel/preset.ts#L77-L195) and use the new `jsx` pragma in React v17 as well as some other transforms NextJS needs. NextJS often updates this preset internals that get shipped with new NextJS versions (minor/major versions), so by just adding `next/babel` you make sure it's always up-to-date with whatever changes NextJS does under the hood.
- renamed to `babel.config.js` because since it's a JS file config, you'd later be able to do:
```javascript
module.exports = () => ({
  presets: ["next/babel"],
  plugins: [
    [
      "babel-plugin-styled-components",
      {
        minify: process.env.NODE_ENV === 'production', // <<< for example
        transpileTemplateLiterals: false,
      },
    ],
  ],
});
```
However, this is out of scope for this PR and it's done mostly to allow more flexibility in the future.